.circleci/config.yml

```diff
@@ -0,0 +1,93 @@
+    version: 2.1
+    executors:
+      docker-publisher:
+        environment:
+          IMAGE_NAME: logdna/logdna-agent
+        docker:
+          - image: circleci/buildpack-deps:stretch
+    jobs:
+      build:
+        executor: docker-publisher
+        steps:
+          - checkout
+          - run: echo "export DATESTAMP=$(date +%Y%m%d)" > env-vars
+          - setup_remote_docker
+          - run:
+              name: Build Docker image
+              command: |
+                docker build -t ${IMAGE_NAME}:latest .
+          - run:
+              name: Archive Docker image
+              command: docker save -o image.tar ${IMAGE_NAME}
+          - persist_to_workspace:
+              root: .
+              paths:
+                - ./image.tar
+                - ./env-vars
+      docker-hub:
+        executor: docker-publisher
+        steps:
+          - attach_workspace:
+              at: /tmp/workspace
+          - run: cat /tmp/workspace/env-vars >> $BASH_ENV
+          - setup_remote_docker
+          - run:
+              name: Load archived Docker image
+              command: docker load -i /tmp/workspace/image.tar
+          - run:
+              name: Publish Docker Image to Docker Hub
+              command: |
+                echo "${DOCKERHUB_PASS}" | docker login -u "${DOCKERHUB_USERNAME}" --password-stdin
+                IMAGE_TAG=${CIRCLE_TAG/v/''}
+                docker tag ${IMAGE_NAME}:latest ${IMAGE_NAME}:${IMAGE_TAG}
```

Comment:
Spit and polish suggestion: Maybe try a for loop instead to reduce duplication?

```
for tag in ${IMAGE_TAG} ${IMAGE_TAG}-${DATESTAMP}; do
  docker tag ${IMAGE_NAME}:latest ${IMAGE_NAME}:${tag}
  docker push ${IMAGE_NAME}:${tag}
done
```

I'd still like to see `docker push ${IMAGE_NAME}:latest` be a step that requires an approval since we dont have testing.  I'd like to see us validate the built image in _our_ deployments before releasing it as latest.

Reply:
in this case, he will need to do `docker push ${IMAGE_NAME}:latest` before the `for` loop. I think instead of `for` loop, comments would be better:
```
# Push with latest tag:
docker push ${IMAGE_NAME}:latest
# Tag and Push with IMAGE_TAG:
docker tag ${IMAGE_NAME}:latest ${IMAGE_NAME}:${IMAGE_TAG}
docker push ${IMAGE_NAME}:${IMAGE_TAG}
# Tag and Push with IMAGE_TAG + DATESTAMP:
docker tag ${IMAGE_NAME}:latest ${IMAGE_NAME}:${IMAGE_TAG}-${DATESTAMP}
docker push ${IMAGE_NAME}:${IMAGE_TAG}-${DATESTAMP} 
```

Reply:
I want to have a conversation about the latest tag. will schedule something.